iroh/src/discovery.rs

```diff
@@ -432,6 +435,47 @@ impl Drop for DiscoveryTask {
     }
 }
 
+/// Error returned when a discovery watch stream lagged too far behind.
+///
+/// The stream returned from [`Endpoint::discovery_stream`] yields this error
+/// if the loop in which the stream is processed cannot keep up with the emitted
+/// discovery events. Attempting to read the next item from the channel afterwards
+/// will return the oldest [`DiscoveryItem`] that is still retained.
+///
+/// Includes the number of skipped messages.
+#[derive(Debug, thiserror::Error)]
+#[error("channel lagged by {0}")]
+pub struct Lagged(pub u64);
+
+#[derive(Clone, Debug)]
+pub(super) struct DiscoverySubscribers {
+    inner: tokio::sync::broadcast::Sender<DiscoveryItem>,
+}
+
+impl DiscoverySubscribers {
+    pub(crate) fn new() -> Self {
+        // TODO: Make capacity configurable from the endpoint builder?
```

Comment:
I'd rather avoid this if possible.  We don't want too many crazy knobs that no one understands in the endpoint builder.